dNet/AuthPackets.cpp

```diff
@@ -64,24 +65,22 @@ void AuthPackets::HandleHandshake(dServer* server, Packet* packet) {
 
 	inStream.IgnoreBytes(33);
 
-	LOG_DEBUG("Client Data [Version: %i, Service: %s, Process: %u, Port: %u, Sysaddr Port: %u]", clientVersion, StringifiedEnum::ToString(serviceId).data(), processID, port, packet->systemAddress.port);
+	LOG_DEBUG("Client Data [Version: %i, Service: %s, Process: %u, Port: %u, Sysaddr Port: %u]", clientVersion, StringifiedEnum::ToString(serviceType).data(), processID, port, packet->systemAddress.port);
 
 	SendHandshake(server, packet->systemAddress, server->GetIP(), server->GetPort(), server->GetServerType());
 }
 
-void AuthPackets::SendHandshake(dServer* server, const SystemAddress& sysAddr, const std::string& nextServerIP, uint16_t nextServerPort, const ServerType serverType) {
+void AuthPackets::SendHandshake(dServer* server, const SystemAddress& sysAddr, const std::string& nextServerIP, uint16_t nextServerPort, const ServiceType serverType) {
 	RakNet::BitStream bitStream;
-	BitStreamUtils::WriteHeader(bitStream, eConnectionType::SERVER, MessageType::Server::VERSION_CONFIRM);
+	BitStreamUtils::WriteHeader(bitStream, ServiceType::COMMON, MessageType::Server::VERSION_CONFIRM);
 
-	const auto clientNetVersionString = Game::config->GetValue("client_net_version");
+	const auto& clientNetVersionString = Game::config->GetValue("client_net_version");
 	const uint32_t clientNetVersion = GeneralUtils::TryParse<uint32_t>(clientNetVersionString).value_or(171022);
 
 	bitStream.Write<uint32_t>(clientNetVersion);
 	bitStream.Write<uint32_t>(861228100);
-
-	if (serverType == ServerType::Auth) bitStream.Write(ServiceId::Auth);
-	else if (serverType == ServerType::World) bitStream.Write(ServiceId::World);
-	else bitStream.Write(ServiceId::General);
+	bitStream.Write(serverType);
+    bitStream.Write<uint16_t>(0); // padding
```

Comment:
```suggestion
	bitStream.Write(static_cast<uint32_t>(serverType));
```
makes more sense than a secondary padding (client also does refer to this as a uint and not a short). This does differ from the original code though in that now General won't be replied if serverType is not Auth or World, though im not sure those servers even handle this packet.

Reply:
Change looks good to me, and I've committed it. Any worries about the potential logic/behavior change? I've playtested the server and haven't noticed any ill effects.